### ISSUE TITLE
[DataGrid] Input fields do not honor defined `theme` variant

### ISSUE THREAD

**Issue by sebastianfrey** [mui/mui-x#2944]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When the `DataGrid` is a child of `ThemeProvider` with a custom theme, which overrides and sets the default `MuiTextField` variant to `outlined`, than the defined variant is not honored by the `DataGrid`s input fields.

### Expected behavior 🤔

Input fields rendered in the context of the `DataGrid` should honor theme overrides.

### Steps to reproduce 🕹

Steps:

1. Go to https://codesandbox.io/s/eager-cray-jz12w?file=/src/Demo.tsx
2. Check the theme: TextField variant is set to `outlined`
2. Click on the filter button or on the columns button.
3. The rendered input fields are not styled with the `outlined` variant.

### Context 🔦

I want to achieve a consistent look and feel in my application for my users.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID 💳 (optional)

22075

**Comment by DanailH** [comment]
After a team discussion, we decided to proceed with option 4. For now, we will include slots for form elements. The naming convention we discussed using is `Base*` so for example the `TextField` will become `BaseTextField`.

**Comment by alexfauquette** [comment]
Sounds this issue has been solved by @DanailH in #3490

The Text fields are customizable with 
```jsx
componentsProps={{
	baseTextField: {
		variant: "outlined",
		margin: "dense",
		size: "small"
	}
}}
```
https://codesandbox.io/s/sleepy-dubinsky-5iuf1o

Can we close this issue, or do we keep it open until we add a slot to each component from `@mui/materil-ui` has a slot?

**Comment by m4theushw** [comment]
Closed by #3490
